test/Constraints/array_literal.swift

```diff
@@ -129,15 +129,15 @@ func defaultToAny(i: Int, s: String) {
 
   let a2: Array = [1, "a", 3.5]
   // expected-error@-1{{heterogeneous collection literal could only be inferred to '[Any]'; add explicit type annotation if this is intentional}}
-  let _: Int = a2  // expected-error{{value of type '[Any]'}}
+  let _: Int = a2  // expected-error{{value of type 'Array<Any>'}}
```

Comment:
It's because we declare 'let a2: Array', so we're only inferring generic arguments here and not the whole type.